src/components/SearchBar.js

```diff
@@ -0,0 +1,62 @@
+import { getSearchData } from "utils/data";
+import { dispatch } from "utils/dispatch";
+
+export class SearchBar {
+  searchBar = document.getElementById("search-bar");
+  autoCompleteElement = document.getElementById("search-bar-autocomplete");
+
+  constructor() {
+    this.searchBar.addEventListener("input", () => {
+      dispatch.call("search-bar-autocomplete", this, this.searchBar.value);
```

Comment:
Unless there's a need for other components to listen to the `"search-bar-autocomplete"` event, you could forgo using `dispatch` here and just bind the callback to the `addEventListener` call. If `this` is giving you trouble for the class method callback you can do either of two things:
- in the constructor do `this.autoComplete` = `this.autoComplete.bind(this)`
- or use an arrow function for defining the class method: 
```js
// within class body:
autoComplete = (value) => {
// ...
} 
```

Reply:
We might also want to **[debounce](https://css-tricks.com/debouncing-throttling-explained-examples/)** the autocomplete in case someone is typing really fast so that we avoid hitting the API too many times (and potentially racking up a bill for MAC).

Reply:
@jimboweb did you see this comment?

Reply:
got rid of the dispatch here

Reply:
The debounce is a good idea and I am going to add that but I don't know how long it'll take to get working so I'd like to make that a separate issue if possible so I can get this live.